deployment/docker-compose.yml

```diff
@@ -0,0 +1,124 @@
+     context: ..
+     dockerfile: ./deployment/Dockerfile
+    healthcheck:
+     timeout: 45s
+     interval: 10s
+     retries: 10
+    environment:
+     - OPENAI_API_KEY=${OPENAI_API_KEY}
+     - WEBSOCKET_URL=${WEBSOCKET_URL:-"ws://localhost/chat"}
+     - CHATBOT_ACCESS_TOKEN=${CHATBOT_ACCESS_TOKEN}
+     - DOMAIN=${DOMAIN:-"localhost:9000"}
+     - POSTGRES_DB_USER=${POSTGRES_DB_USER:-admin}
+     - POSTGRES_DB_PASSWORD=${POSTGRES_DB_PASSWORD:-secret}
+     - POSTGRES_DB_NAME=${POSTGRES_DB_NAME:-adotbcollection}
+     - POSTGRES_DB_HOST=postgresvectordb
+     - POSTGRES_DB_PORT=5432
+    command: uvicorn main:app --host 0.0.0.0 --port 9000 --workers 1
+    logging:
+     options:
+      max-size: 10m
+      max-file: "3"
+    restart: always
+    expose:
+     - 9000
+    # ports:
+    #   - 9000:9000
+    depends_on:
+      - postgresvectordb
+    volumes:
+     - logs-vol:/app/logs
+     - chroma-db:/app/app/chromadb_store
+    networks:
+     - chatbot-network
+
+  postgresvectordb:
+    image: ankane/pgvector
+    ports:
+      - "5435:5432"
+    environment:
+      - POSTGRES_PASSWORD=${POSTGRES_DB_PASSWORD:-secret}
+      - POSTGRES_USER=${POSTGRES_DB_USER:-admin}
+      - POSTGRES_DB=${POSTGRES_DB_NAME:-adotbcollection}
+      - POSTGRES_HOST_AUTH_METHOD=md5
+    restart: always
+    expose:
+      - 5432
+    networks:
+     - chatbot-network
+    volumes:
+     - postgres-db-vol:/var/lib/postgresql/data
+     - postgres-db-backup:/var/backups:rw
+    labels:
+      ofelia.enabled: "true"
+      ofelia.job-exec.backup.schedule: "0 0 0 * * *" #"Takes a single dump within the db container every day at 00:00 am"
+      ofelia.job-exec.backup.command: "pg_dump --dbname=postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@localhost:5432/$POSTGRES_DB  --file=/var/backups/postgres_backup_latest.sql"
+
+  webserver:
+    image: nginx:latest
+    ports:
+     - 80:80
+     - 443:443
+    restart: always
+    expose:
+     - 80
+     - 443
+    depends_on:
+     - chatbot
+    environment:
+      - CHAT_DOMAIN=${DOMAIN:-"localhost"}
+    volumes:
+     - ./nginx/nginx.conf.template:/etc/nginx/templates/default.conf.template:ro
+     # - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+     -  logs-vol:/var/log/nginx/
+     - ./certbot/www:/var/www/certbot/:ro
+     - ./certbot/conf/:/etc/nginx/ssl/:ro
+    networks:
+     - chatbot-network
+
+  certbot:
+    image: certbot/certbot:latest
+    volumes:
+     - ./certbot/www/:/var/www/certbot/:rw
+     - ./certbot/conf/:/etc/letsencrypt/:rw
+    networks:
+     - chatbot-network
+
+  ofelia-scheduler:
+    image: mcuadros/ofelia:v0.3.7
+    depends_on:
+      - postgresvectordb
+    command: daemon --docker
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - postgres-db-backup:/var/backups:r
+      - ./backup:/var/backup-copies/:rw
+    labels:
+      ofelia.job-local.copy-backup-daily.schedule: "0 30 0 * * *" # Copies latest dump from DB contaier every day at 00:30 am
```

Comment:
@kavitharaju I am just wondering about this part:

Takes a single dump within the db container every day at 00:00 am
Copies latest dump from DB container every day at 00:30 am

Once the db get's really big (assuming that happens), is it possible that the dump might not be completed within 30 minutes? Would it make more sense to copy the latest dump just before creating a new dump (i.e., reverse these two steps). That way, a copy is made at 11:59 every day, so there are 24 hours for the copy to complete. It would amount to basically the same thing I suspect. 

This is only a suggestion, not a blocker to the PR, as we could always switch the schedule later if it caused any issues. If the dump happens very quickly then it also would not be a potential future issue as I'm envisioning.

Reply:
If new dump gets created during copy is going on, would that lead to an issue?